Feature/reset depth

leoTiez opened the pull request.

leoTiez committed:
Update: Added command line argument and parameters for learning

learning and testing depth are passed independently now.
Moreover, fixed issues with passing arguments since
training method and constructor of RCNN changed

leoTiez committed:
Update: Made depth dynamic

Use a placeholder for the recurrent depth instead
of setting it in the constructor. This enables
us to use different depth for learning and for testing.

Moreover, fixed typo

bt-s commented on RCNN_tf.py:
```diff
@@ -542,7 +545,7 @@ def train(self, train_data_feats, train_data_labels, val_data_feats,
                     self.input_placeholder: train_data_feats,
                     self.output_placeholder: train_data_labels,
-                    self.num_data_placeholder: batch_size
+                    self.num_data_placeholder: batch_size,
```
Comma could be removed :)

leoTiez replied:
Good point

bt-s commented on RCNN_tf.py:
```diff
@@ -474,6 +476,7 @@ def __init__(
 
     def train(self, train_data_feats, train_data_labels, val_data_feats,
               test_data_feats, test_data_labels, val_data_labels,
+              learning_depth=3, test_depth=6,
```
Personally, I would find ```training_depth``` a more apt naming than ```learning_depth```.

leoTiez replied:
Can do, but it makes it easier for the auto completion. But if you feel happy with it

bt-s commented on RCNN_tf.py:
```diff
@@ -474,6 +476,7 @@ def __init__(
+              learning_depth=3, test_depth=6,
               batch_size=100, epochs=7, dropout_rate=0.2,
               adaptive_learning_factor=.1, performance_update_threshold=3,
               min_difference_accuracy=.005, create_graph=True, print_vars=True,
```
Please describe ```learning_depth``` and ```test_depth``` in the docstring, to keep it consistent and complete.

leoTiez replied:
:+1:

bt-s commented on main.py:
```diff
@@ -200,9 +201,14 @@ def main(argv):
-            if train_args.depth is not None:
-                recurrent_depth_ = train_args.depth
+            if train_args.depth_learning is not None:
+                learning_depth_ = train_args.depth_learning
```
Maybe change ```depth_learning``` to ```learning_depth```.
Idem for ```depth_test``` to ```test_depth```.

leoTiez replied:
Will do

bt-s reviewed (approved):
See comments.

leoTiez committed:
Merge branch 'develop' into feature/resetDepth

# Conflicts:
#	RCNN_tf.py

leoTiez committed:
Update: Changed variable naming

Instead of using learing_depth (test_depth), naming is consistent
with the command line parameters

leoTiez committed:
Update: Minor fixes and docstring

Changed some variable naming, removed a unnecessary comma
and added test_depth and training_depth to the docstrings

leoTiez commented:
Changed the commented passages, will merge now
